sdcm/cluster_aws.py

```diff
@@ -381,6 +381,10 @@ def add_nodes(self, count, ec2_user_data='', dc_idx=0, rack=0, enable_auto_boots
             else:
                 ec2_user_data = post_boot_script
 
+        #  replace user_data json with cloud-init content if preinstalled scylla is not used
```

Comment:
When using separate provision step in Jenkins job, this code won't be executed as initial machines provision goes through different code (https://github.com/scylladb/scylla-cluster-tests/blob/master/sdcm/sct_provision/aws/cluster.py#L239).
From my understanding, this piece of code will be used only for Artifacts test which is not using provision step, so should not affect - this is just for your information.

Reply:
apart from AMI (both artifacts and upgrade) all of them use instances without Scylla pre-installed, so both will use the same mechanism...
the AMI tests do use pre-installed Scylla, so, although it for now doesn't use Ubuntu22, we still need to remember these tests exist.

Reply:
Isn't this affect all OS versions and not only ubuntu 22?

Reply:
yes, but only for artifacts tests (where scylla isn't installed, and it's mostly the ARM ones, that we can't do on GCE)
and @KnifeyMoloko tested it, and proved it to be working (also on multiple OSes, that the changes to sshd config are harmless to older sshd versions)